Automattic/themes
# Blockbase: A different approach to navigation

#### Changes proposed in this Pull Request:
In order to add social links to a navigation block (as suggested in https://github.com/Automattic/themes/pull/4482) using the Customizer, I believe we need a different approach to navigation. One problem with the current approach is that if a user edits their navigation in the Site Editor, edits can no longer be made in Customizer. Another issue is that our approach to the social menu needs significant rework by the user when they switch to the Site Editor.

This PR proposes a different approach:
- We create a new template part called "navigation"
- When a user edits their menus in the Customizer, we recreate the same menu in the template part created above

The major benefit of this approach is that we can build a navigation block for the Site Editor in any way we want. We can use the nav walker class to convert a PHP nav menu to a block nav menu. It's surprisingly simple!

The only downside with this approach is that if I edit the template via the Site Editor, or the Template editor, and edit it again in the customizer, or menus screen, my previous changes would be overwritten.

To test:

- Switch to Blockbase (or any blockbase family theme)
- Create a menu at the "social" location
- Confirm that social links are added to the navigation

Still to do:
- Get the social links to display on the same line as the primary navigation
- Get the preview in customizer to update in real time (this is an issue with the current approach)
